APMrover2/sailboat.cpp

```diff
@@ -174,6 +180,15 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         return;
     }
 
+    // calculate current tack
+    const float app_wind_rad = rover.g2.windvane.get_apparent_wind_direction_rad();
+    const float app_wind_rad_filt = tack_filt.apply(app_wind_rad);
+    if (is_negative(app_wind_rad_filt)) {
+        current_tack = TACK_PORT;
```

Comment:
updating the current_tack here means that calc_heading is dependent upon get_throttle_and_mainsail_out being called first.  I think this will be confusing to future developers because it might be considered odd having the steering behaviour dependent upon the speed controller.  it doesn't look like get_throttle_and_mainsail_out relies on the current_tack.

I think it's possible if this was moved down into calc_heading that the dependency would be removed and I think even the "current_tack" variable could be made local (i..e don't persist it at all).

I think maybe the tack_filt should be run immediately after windvane.update() (perhaps add a new wrapper function in sensors.cpp or sailboat.cpp that includes the windvane.update() and updates this tack_filter) to avoid the possibility that it could be using out of date info.  Either that or the calc_heading will need to be sure the filter is reset if it hasn't been called in a while.

Reply:
we could determine the current tack in the windvane library, its only using wind vane information so i think that would be fair

Reply:
also being in the windvane library we avoid the issue of filtering something that has already filtered, we can use the sensor value directly

Reply:
yes, I agree that it's ok to do the tack decision inside the windvane library.